native/cocos/3d/models/SkinningModel.cpp

```diff
@@ -216,18 +249,105 @@ void SkinningModel::ensureEnoughBuffers(index_t count) {
+        delete _realTimeJointTexture;
+        _realTimeJointTexture = nullptr;
+    }
+    if (pipeline::SkinningJointCapacity::jointUniformCapacity < jointCount) {
+        _realTimeTextureMode = true;
+    }
+    if (!_realTimeTextureMode) return;
+    _realTimeJointTexture = new RealTimeJointTexture;
+    auto *device = gfx::Device::getInstance();
+    uint32_t texWidth = RealTimeJointTexture::WIDTH;
+    uint32_t texHeight = RealTimeJointTexture::HEIGHT;
+    gfx::Format textureFormat = gfx::Format::RGBA32F;
+
+    gfx::FormatFeature formatFeature = device->getFormatFeatures(gfx::Format::RGBA32F);
+    if (!(formatFeature & gfx::FormatFeature::SAMPLED_TEXTURE)) {
+        textureFormat = gfx::Format::RGBA8;
+        texWidth = 4 * RealTimeJointTexture::WIDTH;
+    }
+    uint32_t length = 4 * RealTimeJointTexture::WIDTH * RealTimeJointTexture::HEIGHT;
+    const size_t count = _dataArray.size();
+    for (size_t i = 0; i < count; i++) {
+        delete[] _dataArray[i];
+        _dataArray[i] = new float[length];
+        memset(_dataArray[i], 0, sizeof(float) * length);
+
+        gfx::TextureInfo textureInfo;
+        textureInfo.width = texWidth;
+        textureInfo.height = texHeight;
+        textureInfo.usage = gfx::TextureUsageBit::STORAGE | gfx::TextureUsageBit::SAMPLED | gfx::TextureUsageBit::TRANSFER_SRC | gfx::TextureUsageBit::TRANSFER_DST;
+        textureInfo.format = textureFormat;
+        IntrusivePtr<gfx::Texture> texture = device->createTexture(textureInfo);
+        _realTimeJointTexture->textures.push_back(texture);
+    }
+    _realTimeJointTexture->buffer = new float[length];
+}
+
+void SkinningModel::updateRealTimeJointTextureBuffer() {
+    uint32_t bIdx = 0;
+    uint32_t width = RealTimeJointTexture::WIDTH;
+    uint32_t height = RealTimeJointTexture::HEIGHT;
+    for (const auto &texture : _realTimeJointTexture->textures) {
+        auto *buffer = _realTimeJointTexture->buffer;
+        auto *src    = _dataArray[bIdx];
+        uint32_t count = width;
+        uint32_t index0 = 0;
+        uint32_t index1 = 0;
+        for (uint32_t i = 0; i < count; i++) {
+            index0 = 4 * i;
+            buffer[index0++] = src[index1++];
+            buffer[index0++] = src[index1++];
+            buffer[index0++] = src[index1++];
+            buffer[index0++] = src[index1++];
+            index0 = 4 * (i + width);
+            buffer[index0++] = src[index1++];
+            buffer[index0++] = src[index1++];
+            buffer[index0++] = src[index1++];
+            buffer[index0++] = src[index1++];
+            index0 = 4 * (i + 2 * width);
+            buffer[index0++] = src[index1++];
+            buffer[index0++] = src[index1++];
+            buffer[index0++] = src[index1++];
+            buffer[index0++] = src[index1++];
+        }
+        uint32_t buffOffset = 0;
+        cc::gfx::TextureSubresLayers layer;
+        cc::gfx::Offset texOffset;
+        cc::gfx::Extent extent = {width, height, 1};
```

Comment:
Can just use like
```c++
cc::gfx::Extent extent{width, height, 1};
```

Current implementation will
- create a temp cc::gfx::Extent
- invoke assign constructor
- remove the temp cc::gfx::Extent